crates/sozo/ops/src/keystore.rs

```diff
@@ -85,10 +85,37 @@ pub fn inspect_private(password: Option<String>, raw: bool, file: PathBuf) -> Re
     Ok(())
 }
 
+fn get_password(prompt_text: &str, retry: bool) -> io::Result<String> {
+    let password = rpassword::prompt_password(prompt_text)?;
+
+    if retry {
+        let confirm_password = rpassword::prompt_password("Confirm password: ")?;
+
+        if password == confirm_password {
+            Ok(password)
+        } else {
+            println!("Passwords do not match. Please try again.");
+            get_password(prompt_text, true) // Recursive call for retry
```

Comment:
i don't think we need to repeatedly ask for the confirmation password, if both of the password don't match we can just return an error.

Reply:
You are right